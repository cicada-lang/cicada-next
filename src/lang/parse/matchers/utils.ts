import * as Exps from "../../exp"

export function piBindingtoFnBindingFrom(binding: Exps.PiBinding): Exps.FnBinding {
  switch (binding.kind) {
    case "PiBindingNameless": {
      return Exps.FnBindingName("_")
    }

    case "PiBindingNamed": {
      return Exps.FnBindingName(binding.name)
    }

    case "PiBindingImplicit": {
      return Exps.FnBindingImplicit(binding.name)
    }
  }
}
