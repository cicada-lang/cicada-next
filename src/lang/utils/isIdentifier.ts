const identifierRegExp = new RegExp("^([-_\\p{Letter}][-_\\p{Letter}0-9]*)$", "u")

export function isIdentifier(text: string): boolean {
  return identifierRegExp.test(text)
}
