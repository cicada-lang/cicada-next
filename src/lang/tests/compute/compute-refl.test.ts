import { expect, test } from "vitest"
import { runCode } from "../utils"

test("compute Refl -- under the", async () => {
  const output = await runCode(`

compute the(Equal(String, "abc", "abc"), refl)

`)

  expect(output).toMatchSnapshot()
})

test("compute Refl -- given implicit", async () => {
  const output = await runCode(`

compute refl(implicit String, implicit "abc")

`)

  expect(output).toMatchSnapshot()
})
