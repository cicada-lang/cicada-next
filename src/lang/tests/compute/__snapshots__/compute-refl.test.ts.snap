// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`compute Refl -- given implicit > "refl(implicit String, implicit \\"abc\\"): Equal(String, \\"abc\\", \\"abc\\")" 1`] = `"refl(implicit String, implicit \\"abc\\"): Equal(String, \\"abc\\", \\"abc\\")"`;

exports[`compute Refl -- given implicit: "refl(implicit String, implicit "abc"): Equal(String, "abc", "abc")" 1`] = `"refl(implicit String, implicit "abc"): Equal(String, "abc", "abc")"`;

exports[`compute Refl -- under the > "refl(implicit String, implicit \\"abc\\"): Equal(String, \\"abc\\", \\"abc\\")" 1`] = `"refl(implicit String, implicit \\"abc\\"): Equal(String, \\"abc\\", \\"abc\\")"`;

exports[`compute Refl -- under the: "refl(implicit String, implicit "abc"): Equal(String, "abc", "abc")" 1`] = `"refl(implicit String, implicit "abc"): Equal(String, "abc", "abc")"`;

exports[`compute Refl -- under the 1`] = `"refl(implicit String, implicit "abc"): Equal(String, "abc", "abc")"`;

exports[`compute Refl -- given implicit 1`] = `"refl(implicit String, implicit "abc"): Equal(String, "abc", "abc")"`;
